phpBB/includes/ucp/ucp_main.php

```diff
@@ -213,7 +213,21 @@ function main($id, $mode)
 
 				add_form_key('ucp_front_subscribed');
 
-				$unwatch = (isset($_POST['unwatch'])) ? true : false;
+				$post_data = $_POST;
```

Comment:
It is generally discouraged to use $_POST data. This is only used for checking if it's set. Newer code will use the request class' is_set() and is_set_post() methods. Blindly passing on $_POST data to an event is potentially very dangerous and shouldn't be done. Events should also be able to use the request class themselves.